sockeye/encoder.py

```diff
@@ -108,6 +131,18 @@ def get_rnn_cells(self) -> List[mx.rnn.BaseRNNCell]:
         """
         raise NotImplementedError()
 
+    def get_encoded_data_length(self, data_length: mx.sym.Symbol) -> mx.sym.Symbol:
+        """
+        Returns the data lengths of the encoded sequence.
+        """
+        return data_length
+
+    def get_encoded_seq_len(self, seq_len: int) -> int:
+        """
+        Returns the size of the encoded sequence.
+        """
+        return seq_len
+
```

Comment:
Instead of adding more functions to the interface (that we need to remember to be called), could we return those two pieces of information by default from the encode() method?
That is, change the signature of `encode` to 
```python3
encode(self, data: mx.sym.Symbol, data_length: mx.sym.Symbol, seq_len: int) -> Tuple[mx.sym.Symbol, mx.sym.Symbol,int]:
```

Then most of the encoders just return their encode-input (data_length and seq_len), but the convolutional encoder for example can return modified values. The encoder sequence just returns the last results of the self.encoders.encode() call.